### PR TITLE
RUMM-1313: Support external resource timings

### PR DESCRIPTION
**Pull request by 0xnm**
### What does this PR do?

This PR brings support of external resource timings which may come with `stopResource` call via free-form `attributes` map.

This is needed to support resource timings coming from any consumer of Android Bridge. Code may have a lot of `if`-checks, but this is because we need to read the free-form map, so we don't have advantage of the type system here.

This PR supports the following timings: `ssl`, `connect`, `dns`, `firstByte`, `download`, even though not all may come (like in case of React Native we can only provide `firstByte` and `download`, but this may not be the same for other users of the bridge).

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt [hunk 1]
```diff
@@ -275,12 +278,80 @@ internal class RumResourceScope(
         }
     }
 
+    internal fun tryExtractTimings(timingsPayload: Map<String, Any?>?): ResourceTiming? {
```
Comment (mariusc83) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt:281]: I find it a bit over-engineered. Wouldn't be better just to parse this `Map` as:
```
firstByteStart  ((timingsPayload[FIRST_BYTE_TIMING] as? Map<*,*>)?.get(START_TIME_KEY) as? Number)?.toLong()
```
?
This way we could get rid of all these extra functions and code.

**Review by mariusc83 — CHANGES_REQUESTED** [review]

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt (hunk 1 above)
Comment (0xnm) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt:281]: I tried. I would need to add it 4 times then and it will trigger `ComplexMethod` rule of `Detekt`, need to suppress then.

Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt:281]: I see, that explains why having all these methods.

Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt:281]: LGTM but just as an extra thought if you have time, maybe we could extract all this logic in a separated component ?

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScopeTest.kt [hunk 2]
```diff
@@ -1160,6 +1160,259 @@ internal class RumResourceScopeTest {
         assertThat(resultTiming).isEqualTo(null)
     }
 
+    @Test
```
Comment (mariusc83) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScopeTest.kt:1163]: All these tests are not directly related with the `RumResourceScope` but more with the way we extract the timings from the attributes. That is the reason why I suggested a single responsibility component for that logic.

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt (hunk 1 above)
Comment (0xnm) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt:281]: I agree, probably I will just create Util function for all this (don't want to pollute `Map` namespace with extension function and doubt there is a need to dedicated class).

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumAttributes.kt [hunk 3]
```diff
@@ -85,6 +85,11 @@ object RumAttributes {
      */
     const val SPAN_ID: String = "_dd.span_id"
 
+    /**
+     * Timings coming from external sources, as object { startTime (number) + duration (number) }.
```
[the remark this answers is not in the document]
Comment (0xnm) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumAttributes.kt:89]: yes, makes sense.

**Review by mariusc83 — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]
